pdns/dnssecinfra.cc

```diff
@@ -63,8 +63,13 @@ shared_ptr<DNSCryptoKeyEngine> DNSCryptoKeyEngine::makeFromISCFile(DNSKEYRecordC
   fp.reset();
 
   shared_ptr<DNSCryptoKeyEngine> dke = makeFromISCString(drc, isc);
-  if(!dke->checkKey()) {
-    throw runtime_error("Invalid DNS Private Key in file '"+string(fname));
+  vector<string> checkKeyErrors;
+
+  if(!dke->checkKey(&checkKeyErrors)) {
+    string reason;
+    if(checkKeyErrors.size()) reason = " ("+boost::algorithm::join(checkKeyErrors, ", ")+")";
+    else reason = "";
```

Comment:
Just a nit but I'm not a big fan of the inlined `if` and the `else` clause seems useless.

Reply:
amended